### DISCUSSION TITLE
Livewire breaks mdb dropdown

### DISCUSSION THREAD

**Discussion by ChimpGamer** [livewire/livewire#8053] [1. Help]
Hey,

I'm using Material Design Bootstrap and I've made a dropdown to update a value of a model. But when I use the dropdown to update the value the dropdown no longer works. This is because livewire updates the elements and because of this the `data-mdb-dropdown-initialized="true"` is removed from the button element that opens the actual dropdown. I'm not sure how to work around this. I've already tried things with livewire:ignore but that seems to prevent the button and dropdown menu from updating.

Any help is much appreciated!

Here is a snippet of my code:
```
<div>
    <div class="d-flex ">
        ...
        <div class="dropdown">
            <button type="button" class="btn {{$ticket->priority->buttonStyleClass()}} btn-floating"
                    id="priorityDropDownMenuButton" data-mdb-dropdown-init aria-expanded="false">
                <i class="fa-solid fa-circle-exclamation"></i>
            </button>
            <ul class="dropdown-menu" aria-labelledby="priorityDropDownMenuButton">
                @foreach(\App\Models\Tickets\TicketPriority::cases() as $ticketPriority)
                    <li><button class="dropdown-item" wire:click="setPriority({{$ticketPriority->value}})"><i class="{{$ticketPriority->iconColorClass()}} fa-solid fa-circle-exclamation"></i> {{$ticketPriority->name()}}
                            @if($ticket->priority == $ticketPriority)
                                <i class="fa-solid fa-check float-end"></i>
                            @endif
                        </button></li>
                @endforeach
            </ul>
        </div>
    </div>
    ...
</div>
```

**Comment by danie-ramdhani** [comment]
use `wire:ignore` (https://livewire.laravel.com/docs/wire-ignore)

**Reply by ChimpGamer** [reply]
Like I said: "I've already tried things with livewire:ignore but that seems to prevent the button and dropdown menu from updating."

So that doesn't seem the solution

**Reply by lrljoe** [reply]
It is wire:ignore, not livewire:ignore...

Worth trying wire:ignore.self on the button if that is what is causing the issue

Use distinct wire:key attributes within your for each loops to reduce the work required for DOM assembly , and help Livewire identify existing items during the morphing phase

Please share a wirebox example so that people can investigate properly, as it is not easy to identify issues with only part of your code.

Adding to that, you seem to be doing something funky where you should be passing in the "cases" as a computed property, or as a Livewire Component property 
You should not be calling a model's methids directly via your blade.

**Reply by ChimpGamer** [reply]
Sorry, I wrote this discussion in a bit of a hurry. I meant wire:ignore not livewire:ignore. I've tried it at a couple of place which one of them was like you suggested the button. Adding wire:ignore.self to the button fixes the dropdown not working but breaks the icon of the button itself not updating.
I took your suggestions and applied them to my code. I've no idea what wirebox is and I couldn't find anything about it so I'll be sharing a piece of my code again.
```
<div>
    <div class="d-flex">
        ...
        <div class="dropdown">
            <button wire:ignore.self type="button" class="btn {{$ticket->priority->buttonStyleClass()}} btn-floating"
                    id="priorityDropDownMenuButton" data-mdb-dropdown-init aria-expanded="false">
                <i class="fa-solid fa-circle-exclamation"></i>
            </button>
            <ul class="dropdown-menu" aria-labelledby="priorityDropDownMenuButton">
                @foreach($this->ticketPriorityCases as $ticketPriority)
                    <li wire:key="{{$ticketPriority->value}}">
                        <button class="dropdown-item" wire:click="setPriority({{$ticketPriority->value}})"><i
                                class="{{$ticketPriority->iconColorClass()}} fa-solid fa-circle-exclamation"></i> {{$ticketPriority->name()}}
                            @if($ticket->priority == $ticketPriority)
                                <i class="fa-solid fa-check float-end"></i>
                            @endif
                        </button>
                    </li>
                @endforeach
            </ul>
        </div>
    </div>
    ...
</div>
```

**Reply by lrljoe** [reply]
Yeah, you can't have it both ways, either an element is ignored, or it isn't.

What is mdb dropdown actually doing?  Can you share a screenshot or something, as it's very likely that you can achieve the same with the native AlpineJS

**Reply by ChimpGamer** [reply]
I've made a gif with the provided code. Should be clearer then a screenshot :)
https://i.gyazo.com/06ebf5f4233e92ee1f38c71df8c1bb5e.gif

**Reply by lrljoe** [reply]
Oh that's very much a native alpinejs capable thing, something like this for example should work, it uses the native AlpineJS Anchor plugin that is bundled with Livewire V3:

```
<div x-data="{ open: false }">
    <button x-ref="button" @click="open = ! open"  class="btn {{$ticket->priority->buttonStyleClass()}} btn-floating">
        <i class="fa-solid fa-circle-exclamation"></i>
    </button>
    <div x-show="open" x-anchor="$refs.button">
        <ul class="">
            @foreach($this->ticketPriorityCases as $ticketPriority)
                <li wire:key="{{$ticketPriority->value}}">
                    <button wire:click="setPriority({{$ticketPriority->value}})">
                        <i class="{{$ticketPriority->iconColorClass()}} fa-solid fa-circle-exclamation"></i> 
                        {{ $ticketPriority->name() }}
                        @if($ticket->priority == $ticketPriority)
                            <i class="fa-solid fa-check float-end"></i>
                        @endif
                    </button>
                </li>
            @endforeach
        </ul>
    </div>
</div>
```

You'll need to tweak your styling slightly most likely, but that should at least work in terms of an open/close smoothly

**Reply by ChimpGamer** [reply]
Hmm yea that works indeed without the styling. I'll see if I can get that working somehow.

**Reply by lrljoe** [reply]
AlpineJS does have a ton of native features, always worth considering those over what you're doing currently, and doesn't have the same issues as 3rd party apps that don't expect partial/random page loads.

A lot of 3rd party plugins are designed with traditional "load the page" in mind, not "part of the page changes"

**Reply by ChimpGamer** [reply]
Yea I see but the problem is just styling. I figured out that mdb seems to use popper for the dropdown menu. But I'm not sure how to recreate the exact same dropdown with AlpineJS.
![image](https://github.com/livewire/livewire/assets/19960733/e0410b15-99c2-4dfc-b22f-75dd9c104ac1)

**Reply by lrljoe** [reply]
So Alpine has the anchor directive as demo'd above.

How are you adding this all in, is it all in your app.js?

are you bundling or injecting Livewire/AlpineJS?

A copy of your app.js, and your layout file will be useful if you want to continue with popperjs so that people can try to identify where the issue is.
